packages/mui-material/src/ButtonGroup/ButtonGroup.js

```diff
@@ -81,106 +82,106 @@ const ButtonGroupRoot = styled('div', {
   }),
   [`& .${buttonGroupClasses.grouped}`]: {
     minWidth: 40,
-    '&:not(:first-of-type)': {
-      ...(ownerState.orientation === 'horizontal' && {
-        borderTopLeftRadius: 0,
-        borderBottomLeftRadius: 0,
-      }),
-      ...(ownerState.orientation === 'vertical' && {
-        borderTopRightRadius: 0,
-        borderTopLeftRadius: 0,
+    '&:hover': {
```

Comment:
Nitpick: I noticed this was reorganized, but most styles are unchanged. The reorganization made the git diff much more complex. Do you think it's necessary? Without the restructuring, this would be easier to review.